targets/TARGET_STM/hal_tick_common.c

```diff
@@ -17,9 +17,30 @@
 
 // Overwrite default HAL functions defined as "weak"
 
+// This variable is set to 1 at the of mbed_sdk_init function.
+// The ticker_read_us function must not be called until the mbed_sdk_init is terminated.
+extern int mbed_sdk_inited;
+
+// Variables also reset in HAL_InitTick()
+uint32_t prev_time = 0;
+uint32_t elapsed_time = 0;
+
+// 1 ms tick is required for ST HAL driver
 uint32_t HAL_GetTick()
 {
-    return ticker_read_us(get_us_ticker_data()) / 1000; // 1 ms tick is required for ST HAL
+    uint32_t new_time;
+    if (mbed_sdk_inited) {
+        // Apply the latest time recorded just before the sdk is inited
+        new_time = ticker_read_us(get_us_ticker_data()) + prev_time;
+        prev_time = 0; // Use this time only once
+        return (new_time / 1000);
+    }
+    else {
+        new_time = us_ticker_read();
+        elapsed_time += (new_time - prev_time) & 0xFFFF; // Only use the lower 16 bits
+        prev_time = new_time;
```

Comment:
Is this really needed ?
What about 32b targets ?

Reply:
commit https://github.com/ARMmbed/mbed-os/pull/7352/commits/9523bcf8c97718ed8eb301e1921492c2b04c09fc do this only for 16bits timer